src/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.Ssl.cs

```diff
@@ -579,56 +590,76 @@ internal static void SslSetTargetName(SafeSslHandle sslHandle, string targetName
 
         internal static unsafe void SslCtxSetAlpnProtos(SafeSslHandle ctx, List<SslApplicationProtocol> protocols)
         {
-            byte[] buffer = ConvertAlpnProtocolListToByteArray(protocols);
-            int osStatus;
-
-            fixed (byte* b = buffer)
+            SafeCreateHandle cfProtocolsRefs = null;
+            SafeCreateHandle[] cfProtocolRefs = null;
+            try
             {
-                int result = SslCtxSetAlpnProtos(ctx, (IntPtr)b, buffer.Length, out osStatus);
+               if (protocols.Count == 1 && protocols[0] == SslApplicationProtocol.Http2)
+                {
+                    cfProtocolsRefs = s_cfAlpnHttp211Protocols;
+                }
+                else if (protocols.Count == 1 && protocols[0] == SslApplicationProtocol.Http11)
+                {
+                    cfProtocolsRefs = s_cfAlpnHttp11Protocols;
+                }
+                else if (protocols.Count == 2 && protocols[0] == SslApplicationProtocol.Http2 && protocols[1] == SslApplicationProtocol.Http11)
+                {
+                    cfProtocolsRefs = s_cfAlpnHttp211Protocols;
+                }
+                else
+                {
+                    // we did not match common case. This is more expensive path allocating Core Foundation objects.
+                    cfProtocolRefs = new  SafeCreateHandle[protocols.Count];
+                    IntPtr[] protocolsPtr = new  System.IntPtr[protocols.Count];
```

Comment:
NIt: two spaces after the `new` operator (thought I was having a font or eyeball problem)